mcstatus/motd.py

```diff
@@ -0,0 +1,820 @@
+        if self.ANSI_3_BIT_TRANSFORMER is None:
+            self.ANSI_3_BIT_TRANSFORMER = Ansi3BitTransformer
+        return self.ANSI_3_BIT_TRANSFORMER(self)
+
+    @property
+    def _ansi_4_bit_transformer(self) -> Ansi4BitTransformer:
+        if self.ANSI_4_BIT_TRANSFORMER is None:
+            self.ANSI_4_BIT_TRANSFORMER = Ansi4BitTransformer
+        return self.ANSI_4_BIT_TRANSFORMER(self)
+
+    @property
+    def _ansi_8_bit_transformer(self) -> Ansi8BitTransformer:
+        if self.ANSI_8_BIT_TRANSFORMER is None:
+            self.ANSI_8_BIT_TRANSFORMER = Ansi8BitTransformer
+        return self.ANSI_8_BIT_TRANSFORMER(self)
+
+    @property
+    def _ansi_24_bit_transformer(self) -> Ansi24BitTransformer:
+        if self.ANSI_24_BIT_TRANSFORMER is None:
+            self.ANSI_24_BIT_TRANSFORMER = Ansi24BitTransformer
+        return self.ANSI_24_BIT_TRANSFORMER(self)
+
+    @property
+    def plain(self) -> str:
+        """Get plain text from MOTD, without any colors/formatting.
+
+        :returns: Plain text.
+        """
+        return "".join(str(e) for e in self.parsed if isinstance(e, str))
+
+    @property
+    def minecraft(self) -> str:
+        """Getter for variant of a MOTD with Minecraft colors.
+
+        :returns: Minecraft variant of MOTD.
+
+        .. note:: This will always use "§", even if in original MOTD used "&".
+        """
+        result = ""
+        for element in self.parsed:
+            if isinstance(element, str):
+                result += element
+                continue
+            if isinstance(element, (WebColor, TranslationTag)):
+                continue
+
+            result += "§" + element.value
+
+        return result
+
+    @property
+    def html(self) -> str:
+        """Getter for HTML variant of a MOTD.
+
+        This is just a shortcut for ``HtmlTransformer.format()``.
+        """
+        return self._html_transformer.format()
+
+    @property
+    def ansi_3_bit(self) -> str:
+        """Getter for an ANSI 3-bit colors variant of MOTD.
+
+        This is just a shortcut for ``Ansi3BitTransformer.format()``.
+        """
+        return self._ansi_3_bit_transformer.format()
+
+    @property
+    def ansi_4_bit(self) -> str:
+        """Getter for an ANSI 4-bit colors variant of MOTD.
+
+        This is just a shortcut for ``Ansi4BitTransformer.format()``.
+        """
+        return self._ansi_4_bit_transformer.format()
+
+    @property
+    def ansi_8_bit(self) -> str:
+        """Getter for an ANSI 8-bit colors variant of MOTD.
+
+        This is just a shortcut for ``Ansi8BitTransformer.format()``.
+        """
+        return self._ansi_8_bit_transformer.format()
+
+    @property
+    def ansi_24_bit(self) -> str:
+        """Getter for an ANSI 24-bit colors variant of MOTD.
+
+        This is just a shortcut for ``Ansi24BitTransformer.format()``.
+        """
+        return self._ansi_24_bit_transformer.format()
+
+
+class MotdSimplifies:
```

Comment:
Class doesn't make this very easy either. Overriding can absolutely be achieved by making a child class, overriding the `simplify` method by essentially the same code with a different function, or by monkey patching that function. Indeed this isn't ideal, but it's not impossible.

However I don't see any reason why the user would ever need to override the simplify implementation. If there's something that our logic misses, it should be reported as a bug, not solved locally by overrides.

This could however be simplified if absolutely necessary, by moving the logic of our `simplify` function into `simplify.py` too, and calling it, with the `get_unused_elements` as an argument to it. That way, overriding would be much simpler, by just importing `get_unused_elements`, using that in some other function, which unions it's result with some other custom obtained element IDs. With that, all that would be needed is to make a child Motd class, override the `simplify` method to call the `simplify` function that's now in `simplifies.py`, and pass in a custom function instead of just the `get_unused_elements`.

Then again, I still don't really think that this needs to be "easily overridable", there's no reason anyone should need to override the `simplify` method.

Reply:
Fixed [here](https://github.com/py-mine/mcstatus/compare/c5d0a9f4db39c4962022ae57ab6e295f1faceab0..5761f0d4e4a8c1580071922aa815a849aad2873d).